uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/my-leave/components/my-leave-calendar.js

```diff
@@ -9,21 +11,352 @@ define([
+     *
+     * @param  {int} month
+     * @return {array}
+     */
+    vm.getMonthData = function (month) {
+      if (vm.calendar.days) {
+        var iterator,
+          dates = Object.keys(vm.calendar.days),
+          date,
+          length = dates.length,
+          datesForTheMonth = [];
+
+        for (iterator = 0; iterator < length; iterator++) {
+          date = new Date(parseInt(dates[iterator]));
+          if (date.getMonth() === month) {
+            datesForTheMonth.push(vm.calendar.days[dates[iterator]]);
+          }
+        }
+
+        return datesForTheMonth;
+      }
+    };
+
+    /**
+     * Refresh all leave request and calendar data
+     */
+    vm.refresh = function () {
+      vm.loading = true;
+      loadLeaveRequest()
+        .then(function () {
+          vm.loading = false;
+        });
+    };
+
+    (function init() {
+      vm.loading = true;
+      $q.all([
+        loadAbsencePeriods(),
+        loadAbsenceTypes(),
+        loadPublicHolidays(),
+        loadStatuses(),
+        loadDayTypes()
+      ]).then(function () {
+          vm.legendCollapsed = false;
+          return loadLeaveRequest();
+        })
+        .then(function () {
+          vm.loading = false;
+        });
+    })();
+
+    /**
+     * Returns the leave request which is in range of the sent date
+     *
+     * @param  {string} date
+     * @return {object}
+     */
+    function getLeaveRequest(date) {
+      var iterator,
+        length = leaveRequests.length,
+        dates;
+
+      for (iterator = 0; iterator < length; iterator++) {
+        dates = _.find(leaveRequests[iterator].dates, function (leaveRequestDate) {
+          return moment(leaveRequestDate.date).isSame(date);
+        });
+
+        if (dates) {
+          return leaveRequests[iterator];
+        }
+      }
+    }
+
+    /**
+     * Returns the styles for a specific leaveRequest
+     * which will be used in the view for each date
+     *
+     * @param  {object} leaveRequest
+     * @return {object}
+     */
+    function getStyles(leaveRequest) {
+      var status = leaveRequestStatuses[leaveRequest.status_id],
+        absenceType;
+
+      if (status.name === 'waiting_approval'
+        || status.name === 'approved'
+        || status.name === 'admin_approved') {
+        absenceType = _.find(vm.absenceTypes, function (absenceType) {
+          return absenceType.id == leaveRequest.type_id;
+        });
+
+        return {
+          backgroundColor: absenceType.color,
+          borderColor: absenceType.color
+        };
+      }
+    }
+
+    /**
+     * Returns whether a date is of a specific type
+     * half_day_am or half_day_pm
+     *
+     * @param  {string} name
+     * @param  {object} leaveRequest
+     * @param  {string} date
+     *
+     * @return {boolean}
+     */
+    function isDayType(name, leaveRequest, date) {
+      var dayType = dayTypes[name];
+
+      if (moment(date).isSame(leaveRequest.from_date)) {
```

Comment:
Comparison is necessary, because a leaveRequest can have multiple days inside it, but the half day is possible only for the starting date and the end date. So if the sent date is same as `from_date` then only we should check with `from_day_type`. Similar for `to_date_type`